### PR TITLE
[MLIR][DISC] pattern conversion from tf2mhlo: ConvertTileOpDynamic

### PR DESCRIPTION
**Pull request by azazhu**
We are porting our MLIR-based dynamic shape compiler to tf community (From OP def, Patttern, to Optimization pass, etc).
This is the 4th PR about tf2mhlo pattern conversion, which including ConvertTileOpDynamic.
The rest pattern conversions we will add:
- ConvertSqueezeOpxxx
- ConvertStridedSliceOpxxx
- ConvertUnpackOpxxx
- ConvertPrintOp
- ConvertSigmoidGradOpxxx
- ConvertSignOpxxx

### CODE REVIEW

File: tensorflow/compiler/mlir/xla/transforms/legalize_tf.cc [hunk 1]
```diff
@@ -4236,6 +4236,102 @@ class ConvertTileOp : public OpRewritePattern<TF::TileOp> {
...
+  //   %broadcast_dimensions = [1, 3];
+  //   %broadcast = mhlo.d_broadcast_in_dim(%input, %out_dim_size, %braodcast_dimensions);
+  //   %shape = [MS1, MS2]
+  //   %result = "mhlo.d_reshape"(%broadcast, %shape) : (tensor<S1xM1xS2xM2xf32>) -> tensor<MS1xMS2xf32>
+  // clang-format on
+  LogicalResult matchAndRewrite(
+      TF::TileOp op, PatternRewriter& rewriter) const {
+    auto loc = op.getLoc();
+    auto input = op.input();
+    auto multiples = op.multiples();
```
Comment (joker-eph) [inline tensorflow/compiler/mlir/xla/transforms/legalize_tf.cc:4259]: Please reverse `auto` for when the type is obvious from the immediate context and it improves readability.
(here and elsewhere)

File: tensorflow/compiler/mlir/xla/transforms/legalize_tf.cc [hunk 2]
```diff
@@ -5088,6 +5184,8 @@ class ConvertUnpackOp : public OpRewritePattern<TF::UnpackOp> {
   }
 };
 
+
+
```
Comment (joker-eph) [inline tensorflow/compiler/mlir/xla/transforms/legalize_tf.cc:5188]: Remove these two lines please.

File: tensorflow/compiler/mlir/xla/transforms/legalize_tf.cc (hunk 1 above)
Comment (azazhu) [reply tensorflow/compiler/mlir/xla/transforms/legalize_tf.cc]: Done and I will also refine other PR.

File: tensorflow/compiler/mlir/xla/transforms/legalize_tf.cc (hunk 2 above)
Comment (azazhu) [reply tensorflow/compiler/mlir/xla/transforms/legalize_tf.cc]: Done.

**Review by joker-eph — APPROVED** [review]

**Review by joker-eph — APPROVED** [review]

**Review by rthadur — APPROVED** [review]

**Comment by gbaned** [comment]
@azazhu  Can you please resolve conflicts? Thanks!

**Comment by azazhu** [comment]
> @azazhu Can you please resolve conflicts? Thanks!

Sure, I will resolve this conflict after https://github.com/tensorflow/tensorflow/pull/49919 merged.

**Comment by azazhu** [comment]
> > @azazhu Can you please resolve conflicts? Thanks!
> 
> Sure, I will resolve this conflict after #49919 merged.

Rebased

**Review by joker-eph — APPROVED** [review]

**Pull request merged**